src/mono/wasm/debugger/DebuggerTestSuite/SetVariableValueTests.cs

```diff
@@ -0,0 +1,138 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.WebAssembly.Diagnostics;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace DebuggerTests
+{
+    public class SetVariableValueTests : DebuggerTestBase
```

Comment:
Can we add tests for:

- async cases (static, and instance methods)?
- for primitives:
   - setting strings to null
   - trying to set null for setting non-string types
   - setting int32 value for a int8, etc

- in case of objects:
  - fields: static, instance, readonly
  - properties: static, instance(already have that)
  - unknown property name
  - invalid object id
  - setting a property with no setter
  - set a property that throws an exception
  - setting a property on a derived class instance, which has the setter in the base class, but overrides getter in the derived class
  - setting a property on a derived class instance, via a base class variable, where the derived class overrides the setter

- setting locals on a previous frame, step-check-value, step back to that method, check value
   - and the above for async also

- for primitive locals, can we add tests for all the supported types? You can use TheoryData, and a single test method to do that, so you don't have to lots of repeating.

- We would need tests for CFO types too, but that can be done later.

Reply:
Can we open a issue to create more tests?

Reply:
Sure, but maybe we can add the ones that check for correctness, or invalid inputs?

Reply:
Added.